packages/iTwinUI-react/src/core/Table/columns/actionColumn.tsx

```diff
@@ -6,15 +6,21 @@ import React from 'react';
 import { HeaderProps } from 'react-table';
 import { Checkbox } from '../../Checkbox';
 import SvgColumnManager from '@itwin/itwinui-icons-react/cjs/icons/ColumnManager';
-import { DropdownMenu } from '../../DropdownMenu';
+import { DropdownMenu, DropdownMenuProps } from '../../DropdownMenu';
 import { IconButton } from '../../Buttons/IconButton';
 import { MenuItem } from '../../Menu';
 import { tableResizeStartAction } from '../Table';
 import { SELECTION_CELL_ID } from './selectionColumn';
 import { EXPANDER_CELL_ID } from './expanderColumn';
+import cx from 'classnames';
 
 const ACTION_CELL_ID = 'iui-table-action';
 
+type ActionColumnDropdownMenuProps = Omit<
+  DropdownMenuProps,
+  'menuItems' | 'onHide' | 'onShow' | 'children'
```

Comment:
Ohh, I omitted them here because the `ActionColumn` already specifies the values for `menuItems`, `onHide`, and `onShow`.

![image](https://user-images.githubusercontent.com/45748283/183442165-78d29a1f-e7dc-4508-a489-d6091691179c.png)

So I felt allowing users to override them might cause some unexpected problems since the expected ActionColumn specified functionality may not happen and lead to state inconsistencies because of `ActionColumn` callbacks not being called when expected.

But I can not omit them if they are useful. Or maybe call both the `ActionColumn`'s onHide and onShow along with the user specified onHide and onShow?

Reply:
Yeah, so not override, but extend. You can still keep `isOpen` functionality, but also call users `onHide`/`onShow` if they have provided.
With `menuItems` might be more interesting... Maybe for now we can exclude `menuItems` and `children`. We could add those later, if users need.

Reply:
I've been trying this to do for quite a while, but I don't think I'm understand what's going on behind the scenes to get this to work. I keep getting this error when running `yarn build --filter=storybook`.

```bash
src/core/Table/columns/actionColumn.tsx(121,31): error TS2554: Expected 1 arguments, but got 0.
```

My code looks like this

<!-- ![image](https://user-images.githubusercontent.com/45748283/183477771-a6ff1c84-61b0-47a1-a51e-e5792f5ab52a.png) -->
![image](https://user-images.githubusercontent.com/45748283/183479089-e4f955dd-090b-4f48-9467-6246db5527a9.png)

I do not know what parameter to provide. When I provide a random number as a param, I get this error.

```bash
src/core/Table/columns/actionColumn.tsx(121,40): error TS2345: Argument of type 'number' is not assignable to parameter of type 'Instance<Props>'.
```

`ActionColumnProps` now looks like this

```tsx
type ActionColumnProps = {
  columnManager: boolean | Omit<DropdownMenuProps, 'menuItems' | 'children'>;
};
```

@mayank99 Can you please help?

Reply:
i think you need to forward the parameter:
```ts
onHide={(i) => {
  setIsOpen(false);
  dropdownMenuProps.onHide?.(i);
}}
```

Reply:
**[DONE]** Thanks